syft/rekor/utils.go

```diff
@@ -108,40 +110,42 @@ func validateAttestation(att *InTotoAttestation) error {
 	} else if len(att.Subject) > 1 {
 		err = errors.New("attestation found on rekor contains multiple subjects, which is not supported. Ignoring log entry")
 	} else if _, ok := att.Subject[0].Digest["sha256"]; !ok {
```

Comment:
nit: Just to protect from any incomplete `att` that could be uploaded it might be best to verify the `Subject` at the zero index exists before doing this map check. There also might be an opportunity to exit early in the `len(att.Subject) == 0` case